docs/backend_guide.md

```diff
@@ -0,0 +1,227 @@
+        - DELETE /api/users/{userId}: Delete a user account by ID.
+
+For each endpoint, provide examples of requests and responses along with the expected request and response formats.
+
+
+# Database Schema
+
+    The Power Payment Backend is a secure system for seamless money transfers. Here's an example of the database schema that supports it.
+
+    - User Table:
+
+        user_id (Primary Key)
+        username
+        email
+        password
+        date_created
+
+    - Transaction History Table:
+
+        transaction_history_id (Primary Key)
+        transaction_id (Foreign Key referencing Transaction Table)
+        status (e.g., pending, completed, failed)
+        date_updated
+
+    - Transaction Table:
+
+        transaction_id (Primary Key)
+        sender_id (Foreign Key referencing User Table)
+        receiver_id (Foreign Key referencing User Table)
+        amount
+        date_created
+
+This schema includes tables for users, transactions, and transaction history. The user table stores user information, including username, email, and password. The transaction table tracks payment transactions between users, including sender, receiver, and amount. The transaction history table records the status and update timestamps for each transaction. Keep in mind that this is a basic example, and the actual database schema for the Power Pay App may require additional tables and relationships. 
+
+
+# Authentication
+
+    In the backend, the authentication process for smartphone and Choronko users in the Power Payment system are handled using specific mechanisms and workflows.
+
+        1. Smartphone User Authentication:
+
+            * User Registration: When a smartphone user registers, they provide their phone number, which is stored in the user database.
+            * OTP Generation and Delivery: Upon registration, an OTP is generated and sent to the user's phone number via SMS using an SMS gateway or service provider.
+            * OTP Verification: The backend validates the OTP entered by the user against the one stored in the database. If the OTP matches, the user is considered authenticated.
+            * PIN Setup: After successful OTP verification, the user is prompted to set a PIN for subsequent authentication. The PIN is securely stored in the user database, preferably using hashing and salting techniques.
+            
+        2. Choronko User Authentication:
+
+            * SMS Gateway Integration: The backend integrates with an SMS gateway to receive and interpret messages sent by Choronko users.
+            * Syntax Interpretation: The SMS gateway extracts the phone number and PIN from the incoming message using the specified syntax, such as numberpin#.
+            * PIN Verification: The backend verifies the received PIN against the one stored in the Choronko user's database record. If the PIN matches, the user is considered authenticated.
+
+
+# Error Handling
```

Comment:
I wanted you to discuss these before, at the place you discussed the APIs:

1. The `POST:/api/users` should create users and return 
  - status `200` if everything was alright alongside an object containing the id of the _procedure / registration procedure_, knowing a OTP code would arrive the phone number of the user soon enough...
  - status `400` if the request contains a bad data or...
  - and so on...